src/Umbraco.Cms.Api.Management/Services/Signs/HasPendingChangesSignProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Umbraco.Cms.Api.Management.ViewModels;
+using Umbraco.Cms.Api.Management.ViewModels.Document;
+using Umbraco.Cms.Api.Management.ViewModels.Document.Collection;
+using Umbraco.Cms.Api.Management.ViewModels.Tree;
+using Umbraco.Cms.Core;
+
+
+namespace Umbraco.Cms.Api.Management.Services.Signs;
+
+public class HasPendingChangesSignProvider : ISignProvider
+{
+    private const string Alias = Constants.Conventions.Signs.Prefix + "PendingChanges";
+
+    public bool CanProvideSigns<TItem>()
+        where TItem : IHasSigns =>
+        typeof(TItem) == typeof(DocumentTreeItemResponseModel) ||
+        typeof(TItem) == typeof(DocumentCollectionResponseModel);
+
+    public Task PopulateSignsAsync<TItem>(IEnumerable<TItem> itemViewModels)
+        where TItem : IHasSigns
+    {
+        foreach (TItem item in itemViewModels)
+        {
+            switch (item)
+            {
+                case DocumentTreeItemResponseModel treeItem:
+                    foreach (DocumentVariantItemResponseModel variant in treeItem.Variants)
+                    {
+                        DocumentVariantState state = variant.State;
+                        if (state == DocumentVariantState.PublishedPendingChanges)
```

Comment:
```suggestion
                        if (variant.State == DocumentVariantState.PublishedPendingChanges)
```

Just as we don't use this variable again, so doesn't add anything really to declare it and not just use the `State` property directly.